app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/integration/IntegrationOverviewHelper.java

```diff
@@ -84,48 +86,50 @@ static IntegrationOverview toCurrentIntegrationOverview(final Integration integr
                 .collect(Collectors.toList()));
         }
 
-        IntegrationDeployment deployed = null;
+        final ListResult<IntegrationDeployment> deployments = dataManager.fetchAll(IntegrationDeployment.class,
+                new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance());
+
         final List<IntegrationDeployment> activeDeployments = new ArrayList<>();
-        for (final IntegrationDeployment deployment : dataManager.fetchAll(IntegrationDeployment.class,
-            new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance())) {
+        for (final IntegrationDeployment deployment : deployments) {
             builder.addDeployment(IntegrationDeploymentOverview.of(deployment));
 
-            final IntegrationDeploymentState currentState = deployment.getCurrentState();
-            if (currentState == IntegrationDeploymentState.Published) {
-                deployed = deployment;
-                builder.deploymentVersion(deployment.getVersion());
-            }
-
-            if (currentState != IntegrationDeploymentState.Unpublished) {
-                // the bet is that any integration that the user wanted to
-                // publish
-                // will have it's status != Unpublished, the reason why we can't
-                // look at the last deployment is because users can choose to
-                // deploy
-                // previous deployments, so we bet that all the Unpublished
-                // integrations are not the ones that the user don't hold the
-                // current state
-                builder.targetState(deployment.getTargetState());
-                builder.currentState(currentState);
+            if (deployment.getCurrentState().equals(IntegrationDeploymentState.Pending) ||
+                deployment.getCurrentState().equals(IntegrationDeploymentState.Published)) {
                 activeDeployments.add(deployment);
             }
         }
 
-        if (deployed != null) {
-            builder.isDraft(computeDraft(integration, deployed.getSpec()));
+        Optional<IntegrationDeployment> maybePublished = activeDeployments.stream()
+                  .filter(deployment -> deployment.getCurrentState().equals(IntegrationDeploymentState.Published))
+                  .findAny();
+
+        IntegrationDeployment exposedDeployment = null;
+        if (maybePublished.isPresent()) {
+            IntegrationDeployment published = maybePublished.get();
+            builder.isDraft(computeDraft(integration, published.getSpec()));
+            builder.version(published.getVersion());
+            exposedDeployment = published;
+        } else if (activeDeployments.size() == 1) {
+            exposedDeployment = activeDeployments.get(0);
```

Comment:
Not 100% sure about correctness as it also might contain integrations in `Pending` state

Reply:
I think we should include pending here as it can be any deployment in the list due to redeploy functionality. The UI then displays the current status of the pending deployment.

Publish state wins 1st then any pending deployment then the latest deployment regardless its current state